courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst

```diff
@@ -0,0 +1,802 @@
+The C Developer's Perspective on Ada
+======================================
+
+What we mean by Embedded Software
+------------------------------------
+The Ada programming language is a general programming language, which means it can be used for many different types of applications. One type of application where it particularly shines is reliable and safety-critical embedded software; meaning, a platform with a microprocessor such as ARM, PowerPC, x86, or RISC-V. The application may be running on top of an embedded operating system, such as an embedded Linux, or directly on bare metal. And the application domain can range from small entities such as firmware or device controllers to flight management system, communication based train control systems, or advanced driver assistance systems.
+
+The GNAT Toolchain
+-------------------
+
+The toolchain used throughout this book is called GNAT, which is a suite of tools with a compiler based on the GCC environment. It can be obtained from AdaCore, either as part of a commercial contract with `GNAT Pro <https://www.adacore.com/gnatpro)>`_ or at no charge with the `GNAT Community edition <https://www.adacore.com/community>`_. The information on this book will be relevant no matter which edition you’re using. Most examples will be runnable on the native Linux or Windows version for convenience. Some will only be relevant in the context of a cross toolchain, in which case we’ll be using the embedded ARM bare metal toolchain.
```

Comment:
We might want to actually remove this section completely since we aren't asking the user to run anything on their system. I don't think its necessary to mention the FSF since we wouldn't be testing the code snippets on the FSF.